### PR TITLE
Compactify rotation matrix build

### PR DESCRIPTION
**Pull request by dschwen**
Refs #16016

### CODE REVIEW

**Comment by dschwen** [comment]
Oddly enough this will probably decrease net coverage by reducing the number of tested LOC :-)

File: modules/tensor_mechanics/src/materials/HillConstants.C [hunk 1]
```diff
@@ -184,47 +184,19 @@ HillConstants::rotateHillConstants(const std::vector<Real> & hill_constants_inpu
...
-  transformation_matrix_n(4, 5) = trm(0, 0) * trm(2, 1) + trm(0, 1) * trm(2, 0);
-
-  transformation_matrix_n(5, 0) = 2.0 * trm(0, 0) * trm(1, 0);
-  transformation_matrix_n(5, 1) = 2.0 * trm(0, 1) * trm(1, 1);
-  transformation_matrix_n(5, 2) = 2.0 * trm(0, 2) * trm(1, 2);
-  transformation_matrix_n(5, 3) = trm(0, 1) * trm(1, 2) + trm(0, 2) * trm(1, 1);
-  transformation_matrix_n(5, 4) = trm(0, 0) * trm(1, 2) + trm(0, 2) * trm(1, 0);
-  transformation_matrix_n(5, 5) = trm(0, 0) * trm(1, 1) + trm(0, 1) * trm(1, 0);
+
+  // construct "bond-like" rotation matrix
```
Comment (recuero) [inline modules/tensor_mechanics/src/materials/HillConstants.C:188]: Where is the bond-like transformation matrix documented?

Comment (dschwen) [reply modules/tensor_mechanics/src/materials/HillConstants.C:188]: I cannot find where I read "bond-like" 🔫 , [this discussion](https://en.wikiversity.org/wiki/Talk:Coordinate_transformations) calls it _bound matrix_, but Google did not give me any useful definition of that either. I'll just remove the comment. 

Either way, the loop sets up exactly the same matrix that the unrolled version set up.

Comment (dschwen) [reply modules/tensor_mechanics/src/materials/HillConstants.C]: Oh, I found it, "bond-like" is what it is called in https://scicomp.stackexchange.com/questions/35600/4th-order-tensor-rotation-sources-to-refer which is the link you added in the comment above...

**Comment by recuero** [comment]
Are you adding shell inertia coverage in this PR?

File: modules/tensor_mechanics/src/materials/HillConstants.C (hunk 1 above)
Comment (recuero) [reply modules/tensor_mechanics/src/materials/HillConstants.C]: So, basically you did the math to condense the number of lines needed to compute the transformation matrix for compliance tensors, right?

**Comment by dschwen** [comment]
> Are you adding shell inertia coverage in this PR?

That should be a different branch/PR

File: modules/tensor_mechanics/src/materials/HillConstants.C (hunk 1 above)
Comment (dschwen) [reply modules/tensor_mechanics/src/materials/HillConstants.C]: yes, that's all there is to this

**Review by recuero — APPROVED** [review]

**Review by cticenhour — APPROVED** [review]

**Pull request merged**